hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java

```diff
@@ -52,31 +62,50 @@
                 grpcChannelRegistry.forAddress(
                     serviceConfig.getGatewayServiceHost(), serviceConfig.getGatewayServicePort()))
             .withCallCredentials(credentials);
+
+    this.labelConfigServiceStub =
+        LabelsConfigServiceGrpc.newFutureStub(
+                grpcChannelRegistry.forAddress(
+                    serviceConfig.getConfigServiceHost(), serviceConfig.getConfigServicePort()))
+            .withCallCredentials(credentials);
   }
 
   @Override
   public Single<EntityResultSet> getEntities(EntityRequest request) {
-    return this.requestBuilder
-        .buildRequest(request)
-        .subscribeOn(this.boundedIoScheduler)
-        .flatMap(
-            serverRequest ->
-                this.makeRequest(request.resultSetRequest().context(), serverRequest)
-                    .flatMap(
-                        serverResponse ->
-                            baselineDao
-                                .getBaselines(
-                                    request.resultSetRequest().context(),
-                                    serverRequest,
-                                    serverResponse,
-                                    request)
-                                .flatMap(
-                                    baselineResponse ->
-                                        this.entityConverter.convert(
-                                            request, serverResponse, baselineResponse))));
+    return Single.zip(
+            this.requestBuilder
```

Comment:
We can actually do the following instead of making the server request and response a pair

```
Single<EntitiesRequest> entitiesRequestSingle = this.requestBuilder.buildRequest(request);
return Single.zip(
  entitiesRequestSingle,
  entitiesRequestSingle..subscribeOn(this.boundedIoScheduler).flatMap(....), // no need to make a pair now
  this.makeLabelRequest(request.resultSetRequest().context(), request)
                .subscribeOn(this.boundedIoScheduler),
  (entitiesRequest, entitiesResponse, labelsResponse) -> {.....})
```

Reply:
This one looks much cleaner. Previous one was your input as well ;-)